src/app/features/profile/phone-verification/phone-verification.page.ts

```diff
@@ -0,0 +1,186 @@
+                  {
+                    key: 'phoneNumber',
+                    type: 'input',
+                    templateOptions: {
+                      label: enterPhoneNumberTranslation,
+                      type: 'tel',
+                      placeholder: phonePlaceHolderTranslation,
+                      required: true,
+                      pattern: TEL_REGEXP,
+                    },
+                  },
+                ],
+              },
+            ];
+            this.verificationCodeFields = [
+              {
+                fieldGroup: [
+                  {
+                    key: 'verificationCode',
+                    type: 'input',
+                    templateOptions: {
+                      label: enterVerificationCodeTranslation,
+                      type: 'tel',
+                      placeholder: verificationCodePlaceHolderTranslation,
+                      required: true,
+                      pattern: VERIFICATION_CODE_REGEXP,
+                    },
+                  },
+                ],
+              },
+            ];
+          }
+        ),
+        untilDestroyed(this)
+      )
+      .subscribe();
+  }
+
+  onPhoneNumberFormSubmit() {
+    const ONE_SECOND = 1000;
+    const action$ = this.diaBackendAuthService
+      .sendPhoneVerification$(this.phoneNumberModel.phoneNumber)
+      .pipe(
+        catchError((err: unknown) => this.errorService.toastError$(err)),
+        tap(() => {
+          this.hasSentPhoneVerification = true;
+          timer(0, ONE_SECOND)
+            .pipe(
+              take(this.RESEND_INTERVAL),
+              map(i => this.RESEND_INTERVAL - i - 1)
+            )
+            .subscribe(val => {
+              this.secondsRemained = val;
+            });
+        })
+      );
+    return this.blockingActionService
+      .run$(action$, {
+        message: this.translocoService.translate('message.pleaseWait'),
+      })
+      .pipe(untilDestroyed(this))
+      .subscribe();
+  }
+
+  onVerificationCodeFormSubmit() {
+    const THREE_SECONDS = 3000;
+    const action$ = this.diaBackendAuthService
+      .verifyPhoneVerification$(
+        this.phoneNumberModel.phoneNumber,
+        this.verificationCodeModel.verificationCode
+      )
+      .pipe(
+        catchError((err: unknown) => this.handleVerificationError$(err)),
+        tap(() => {
+          this.snackBar.open(
+            this.translocoService.translate('message.verificationSuccess')
+          );
+          timer(THREE_SECONDS)
+            .pipe(first())
+            .subscribe(() => {
+              return this.router.navigate(['..'], {
+                relativeTo: this.route,
+              });
+            });
+        })
+      );
+    return this.blockingActionService
+      .run$(action$, {
+        message: this.translocoService.translate('message.pleaseWait'),
+      })
+      .pipe(untilDestroyed(this))
+      .subscribe();
+  }
+
+  private handleVerificationError$(err: unknown) {
```

Comment:
There is also a `throttled` error that could possibly happen when the user is sending verification phone number or code over 5 times per minute. This is achievable by leaving the page and go back to the page again (the current button lock status and cooldown would not persist after component destruction).

The priority is low, however, since it only happens when user tries to intentionally abuse the mechanism of App, and the effect is not permanent. (after 1 minute the API will stop responding will `throttled` error) Most user would probably never meet this API error.